crates/sui-core/src/execution_cache/writeback_cache.rs

```diff
@@ -458,6 +472,80 @@ impl WritebackCache {
         )
     }
 
+    fn get_transaction_lock(
+        &self,
+        obj_ref: &ObjectRef,
+        epoch_store: &AuthorityPerEpochStore,
+    ) -> SuiResult<Option<LockDetails>> {
+        match self.locked_transactions.entry(*obj_ref) {
+            DashMapEntry::Vacant(vacant) => {
+                let tables = epoch_store.tables()?;
+                let lock = tables.get_locked_transaction(obj_ref)?;
+                if let Some(lock_details) = lock {
+                    vacant.insert(lock_details);
+                }
+                Ok(lock)
+            }
+            DashMapEntry::Occupied(occupied) => Ok(Some(*occupied.get())),
```

Comment:
Would it be useful to add a `debug_assert` that the lock must match the db if it's in the cache?

Reply:
done